contracts/gauge/Gauge.sol

```diff
@@ -0,0 +1,217 @@
+        require(amount > 0, "Cannot stake 0");
+        _totalSupply = _totalSupply.add(amount);
+        _balances[account] = _balances[account].add(amount);
+        emit Staked(account, amount);
+        TOKEN.safeTransferFrom(account, amount);
+    }
+
+    function depositAll() external {
+        _deposit(TOKEN.balanceOf(msg.sender), msg.sender);
+    }
+
+    function deposit(uint256 amount) external {
+        _deposit(amount, msg.sender);
+    }
+
+    function depositFor(uint256 amount, address account) external {
+        _deposit(amount, account);
+    }
+
+    function _withdraw(uint amount) internal nonReentrant updateReward(msg.sender) {
+        require(amount > 0, "Cannot withdraw 0");
+        _totalSupply = _totalSupply.sub(amount);
+        _balances[msg.sender] = _balances[msg.sender].sub(amount);
+        TOKEN.safeTransfer(msg.sender, amount);
+        emit Withdrawn(msg.sender, amount);
+    }
+
+    function withdrawAll() external {
+        _withdraw(_balances[msg.sender]);
+    }
+    
+    function withdraw(uint256 amount) external {
+        _withdraw(amount);
+    }
+
+    function getReward(uint256 tokenIndex) public nonReentrant updateReward(msg.sender) {
+        uint256 reward = rewards[msg.sender][rewardTokens[tokenIndex]];
+        if (reward > 0) {
+            rewards[msg.sender][rewardTokens[tokenIndex]] = 0;
+            AXIAL.safeTransfer(msg.sender, reward);
+            emit RewardPaid(msg.sender, reward, tokenIndex);
+        }
+    }
+
+    function exit() external {
+       _withdraw(_balances[msg.sender]);
+       for (uint256 i = 0; i < rewardTokens.length; i++){
+           getReward(i);
+       }
+        
+    }
+
+    function notifyRewardAmount(uint256 reward, uint tokenIndex) external onlyDistribution updateReward(address(0)) {
+        AXIAL.safeTransferFrom(DISTRIBUTION, reward);
+        if (block.timestamp >= periodFinish) {
+            rewardRates[rewardTokens[tokenIndex]] = reward.div(DURATION);
+        } else {
+            uint256 remaining = periodFinish.sub(block.timestamp);
+            uint256 leftover = remaining.mul(rewardRates[rewardTokens[tokenIndex]]);
+            rewardRates[rewardTokens[tokenIndex]] = reward.add(leftover).div(DURATION);
+        }
+
+        // Ensure the provided reward amount is not more than the balance in the contract.
+        // This keeps the reward rate in the right range, preventing overflows due to
+        // very high values of rewardRate in the earned and rewardsPerToken functions;
+        // Reward + leftover must be less than 2^256 / 10^18 to avoid overflow.
+        uint balance = AXIAL.balanceOf(address(this));
+        require(rewardRates[rewardTokens[tokenIndex]] <= balance.div(DURATION), "Provided reward too high");
+
+        lastUpdateTime = block.timestamp;
+        periodFinish = block.timestamp.add(DURATION);
+        emit RewardAdded(reward);
+    }
+   
+        
+    
+}
```

Comment:
We should update all function headers to include the natspec formatted comment header linked by Bayological.  The header should at the very least describe *why* we are doing something, define the parameters and return values.  If it is in any way unclear *what* the function is doing, the *what* should be included along with the *why*.  If the variable and function names are really well-thought-out, the *what* may not be needed.